arch/x86/core/intel64/userspace.S

```diff
@@ -326,16 +331,13 @@ z_x86_userspace_enter:
 	cli
 
 #ifdef CONFIG_EXECUTION_BENCHMARKING
-	/* Save the eax and edx registers before reading the time stamp
-	* once done pop the values.
-	*/
-	push %rax
-	push %rdx
-	rdtsc
-	mov %eax,arch_timing_enter_user_mode_end
-	mov %edx,arch_timing_enter_user_mode_end+4
-	pop %rdx
-	pop %rax
+	push	%rax
```

Comment:
I don't think RAX needs preservation, note the 'movq' right below this.
RDX needs to be saved, it has the 'p2' argument